core/src/test/java/com/google/cloud/sql/core/CoreSocketFactoryTest.java

```diff
@@ -347,31 +346,119 @@ public void create_notAuthorizedToGetInstance() throws IOException {
+   *      "code":403,
+   *      "message":"Cloud SQL Admin API has not been used in project ### before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=### then retry. If you enabled this API recently, wait a few minutes for the action to  propagate to our systems and retry.",
+   *      "errors":[
+   *         {
+   *            "message":"Cloud SQL Admin API has not been used in project ### before or it is disabled. Enable it by visiting  https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=###then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.",
+   *            "domain":"usageLimits",
+   *            "reason":"accessNotConfigured",
+   *            "extendedHelp":"https://console.developers.google.com"
+   *         }
+   *      ],
+   *      "status":"PERMISSION_DENIED"
+   *   }
+   * }
+   */
+  private GoogleJsonResponseException fakeNotConfiguredException() throws IOException {
+    return fakeGoogleJsonResponseException(
+        HttpStatusCodes.STATUS_CODE_FORBIDDEN,
+        "accessNotConfigured",
+        "Cloud SQL Admin API has not been used in project 12345 before or it is disabled. Enable"
+            + " it by visiting "
+            + " https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=12345"
+            + " then retry. If you enabled this API recently, wait a few minutes for the action to"
+            + " propagate to our systems and retry.");
+  }
+
+  /*
+   * Creates a fake "notAuthorized" exception that can be used for testing.
+   *
+   * Example Response:
+   *  {
+   *   "error":{
+   *      "code":403,
+   *      "message":"The client is not authorized to make this request.",
+   *      "errors":[
+   *         {
+   *            "message":"The client is not authorized to make this request.",
+   *            "domain":"global",
+   *            "reason":"notAuthorized"
+   *         }
+   *      ]
+   *   }
+   *  }
+   */
+  private GoogleJsonResponseException fakeNotAuthorizedException() throws IOException {
+    return fakeGoogleJsonResponseException(
+        HttpStatusCodes.STATUS_CODE_FORBIDDEN,
+        "notAuthorized",
+        "The client is not authorized to make this request");
+  }
+
+  /** Builds a fake GoogleJsonResponseException for testing API error handling. */
+  private static GoogleJsonResponseException fakeGoogleJsonResponseException(
+      int httpStatus, String reason, String message) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+    errorInfo.setReason(reason);
+    errorInfo.setMessage(message);
+    return fakeGoogleJsonResponseException(httpStatus, errorInfo, message);
+  }
+
+  private static GoogleJsonResponseException fakeGoogleJsonResponseException(
+      int status, ErrorInfo errorInfo, String httpStatusString) throws IOException {
```

Comment:
Should the parameter be called "httpStatusString"? It's called "message" in the callers. "httpStatusString" makes me think of the "OK" in "200 OK", but we're not using it like that.